lib/spack/spack/test/cmd/compiler_command.py

```diff
@@ -0,0 +1,38 @@
+# Copyright 2013-2019 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+import pytest
+
+import os
+import os.path
+
+import spack.main
+import spack.compilers
+
+compiler = spack.main.SpackCommand('compiler')
+
+
+@pytest.fixture
+def no_compilers_yaml(mutable_config, monkeypatch):
+    """Creates a temporary configuration without compilers.yaml"""
+
+    for scope, local_config in mutable_config.scopes.items():
+        compilers_yaml = os.path.join(
+            local_config.path, scope, 'compilers.yaml'
+        )
+        if os.path.exists(compilers_yaml):
+            os.remove(compilers_yaml)
+
+    # This is essential, otherwise the cache will create weird side effects
+    # that will compromise subsequent tests
+    monkeypatch.setattr(spack.compilers, '_cache_config_file', [])
```

Comment:
I think when you say

> caches at module level make it much more difficult to diagnose test failures - maybe we could have a look how to rework them sooner or later

This is what you are referring to? IMO it would be worth specifying a TODO here

Also, it may be worth moving this to the `mutable_config` fixture (ideally that would handle everything to return the configuration to its original state, including side effects from caching functions).

Reply:
Done. Let me know if the TODO is fine or it should be rephrased somehow.